packages/block-editor/src/components/global-styles/dimensions-panel.js

```diff
@@ -0,0 +1,622 @@
+	__experimentalVStack as VStack,
+	__experimentalUnitControl as UnitControl,
+	__experimentalUseCustomUnits as useCustomUnits,
+	__experimentalView as View,
+} from '@wordpress/components';
+import { Icon, layout, positionCenter, stretchWide } from '@wordpress/icons';
+import { useCallback, Platform } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getValueFromVariable } from './utils';
+import SpacingSizesControl from '../spacing-sizes-control';
+import HeightControl from '../height-control';
+import ChildLayoutControl from '../child-layout-control';
+import { cleanEmptyObject } from '../../hooks/utils';
+
+const AXIAL_SIDES = [ 'horizontal', 'vertical' ];
+
+export function useHasDimensionsPanel( settings ) {
+	const hasContentSize = useHasContentSize( settings );
+	const hasWideSize = useHasWideSize( settings );
+	const hasPadding = useHasPadding( settings );
+	const hasMargin = useHasMargin( settings );
+	const hasGap = useHasGap( settings );
+	const hasMinHeight = useHasMinHeight( settings );
+	const hasChildLayout = useHasChildLayout( settings );
+
+	return (
+		Platform.OS === 'web' &&
+		( hasContentSize ||
+			hasWideSize ||
+			hasPadding ||
+			hasMargin ||
+			hasGap ||
+			hasMinHeight ||
+			hasChildLayout )
+	);
+}
+
+function useHasContentSize( settings ) {
+	return settings?.layout?.contentSize;
+}
+
+function useHasWideSize( settings ) {
+	return settings?.layout?.wideSize;
+}
+
+function useHasPadding( settings ) {
+	return settings?.spacing?.padding;
+}
+
+function useHasMargin( settings ) {
+	return settings?.spacing?.margin;
+}
+
+function useHasGap( settings ) {
+	return settings?.spacing?.blockGap;
+}
+
+function useHasMinHeight( settings ) {
+	return settings?.dimensions?.minHeight;
+}
+
+function useHasChildLayout( settings ) {
+	const {
+		type: parentLayoutType = 'default',
+		default: { type: defaultParentLayoutType = 'default' } = {},
+		allowSizingOnChildren = false,
+	} = settings?.parentLayout ?? {};
+
+	const support =
+		( defaultParentLayoutType === 'flex' || parentLayoutType === 'flex' ) &&
+		allowSizingOnChildren;
+
+	return !! settings?.layout && support;
+}
+
+function useHasSpacingPresets( settings ) {
+	const {
+		custom,
+		theme,
+		default: defaultPresets,
+	} = settings?.spacing?.spacingSizes || {};
+	const presets = custom ?? theme ?? defaultPresets ?? [];
+
+	return presets.length > 0;
+}
+
+function filterValuesBySides( values, sides ) {
+	if ( ! sides ) {
+		// If no custom side configuration all sides are opted into by default.
+		return values;
+	}
+
+	// Only include sides opted into within filtered values.
+	const filteredValues = {};
+	sides.forEach( ( side ) => {
+		if ( side === 'vertical' ) {
+			filteredValues.top = values.top;
+			filteredValues.bottom = values.bottom;
+		}
+		if ( side === 'horizontal' ) {
+			filteredValues.left = values.left;
+			filteredValues.right = values.right;
+		}
+		filteredValues[ side ] = values[ side ];
```

Comment:
This appears to be the line that's throwing an error when resetting an individual margin value at the block level with the Group block. Possibly a related note here is that the Group block's opt in for margin only includes the top/bottom sides: `"margin": [ "top", "bottom" ]`.

![image](https://user-images.githubusercontent.com/14988353/220804386-dd3e371f-64b0-474d-889f-146ccda1bcf1.png)

Reply:
Looks like this line was fixed in https://github.com/WordPress/gutenberg/pull/48070/commits/d055ea5dbf5553a5d6daa72b5ca9c440ba8231c1 and then possibly accidentally reverted in https://github.com/WordPress/gutenberg/pull/48070/commits/e46b52edf191726d80d1da9b97d28f5acab7be92? Resetting margin is currently causing the block error, but using the optional chaining fix from https://github.com/WordPress/gutenberg/pull/48070/commits/d055ea5dbf5553a5d6daa72b5ca9c440ba8231c1  fixes it for me 🙂

Reply:
🤕